solc/CommandLineInterface.cpp

```diff
@@ -1190,9 +1190,11 @@ void CommandLineInterface::outputCompilationResults()
 	if (!m_hasOutput)
 	{
 		if (!m_options.output.dir.empty())
+		{
 			sout() << "Compiler run successful. Artifact(s) can be found in directory " << m_options.output.dir << "." << endl;
-		else
-			serr() << "Compiler run successful, no output requested." << endl;
+		}
+		else if (contracts.empty())
+			sout() << "Compiler run successful, no contracts to compile." << endl;
```

Comment:
What about the case where there is no output but there are some contracts?

If that's not possible, you should assert that it's not. If it is, you should handle it.

Reply:
I have added a case for catching instances where there are contracts but no output.